src/main/java/useCase/LocationFilterType.java

```diff
@@ -8,17 +8,20 @@
 public class LocationFilterType implements FilterType {
     @Override
     public UserAccount[] filter() {
+        // Create database and get info from the current user
         DatabaseManager databaseManager = DatabaseManager.getDatabaseManager();
-        String[] username = databaseManager.list_username();
+        String[] username = databaseManager.listUsername();
         ArrayList<UserAccount> match= new ArrayList<>();
-        CurrUser curr = CurrUser.getCurrUser();
-        UserAccount currUser = databaseManager.retrieve_user_account(curr.getCurrUsername());
+        CurrUserManager currUserManager = CurrUserManager.getCurrUserManager();
+        UserAccount currUser = currUserManager.getCurrUser();
+        // Check the location of every user in the database, only store people who
+        // in the same province or same city with the current user (same city first)
         for (String user : username) {
-            UserAccount account = databaseManager.retrieve_user_account(user);
-            if (Objects.equals(account.get_country(), currUser.get_country()) &
-                    !Objects.equals(account.get_username(), currUser.get_username())) {
-                if (Objects.equals(account.get_province(), currUser.get_province())) {
-                    if (Objects.equals(account.get_city(), currUser.get_city())){
+            UserAccount account = databaseManager.retrieveUserAccount(user);
+            if (Objects.equals(account.getCountry(), currUser.getCountry()) &
+                    !Objects.equals(account.getUsername(), currUser.getUsername())) {
```

Comment:
You may want to simplify this if statement or extract it to a method (use IntelliJ's Extract Method feature) to make this a bit easier to read.